### DISCUSSION TITLE
MAUI RC1 Migration - Unable to compile Android

### DISCUSSION THREAD

**Discussion by Cywizz** [dotnet/maui#6048] [Q&A] [answered]
Hi,
I started a MAUI project with the previous preview. I migrated the project to RC1.
I followed the migration guide for RC1 and removed the following line from my csproj:

```
<ItemGroup Condition="$(TargetFramework.Contains('-windows'))">
  <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.1" />
  <PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.2" />
</ItemGroup>
```

The only references left are these:
```
<ItemGroup>
    <PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.1" />
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.2" />
</ItemGroup>
```

Windows works fine, however I am unable to compile to Android and getting the following error:
**_Microsoft.Graphics.Canvas.winmd cannot be referenced. Referencing a Windows Metadata component directly when targeting .NET 5 or higher is not supported._** 

The previous output was a warning that a copy failed:
**_Microsoft.Graphics.Canvas.dll could not be copied because the AnyCPU platform is being used. Please specify a specific platform to copy this file._**	

Is there something else I should change somewhere that I am missing? Must I recreate the emulator perhaps? Thanks in advance

**Comment by Cywizz** [comment, marked as answer]
Never mind...Not sure why I left the general reference in there also...deleting it solved the issue

**Comment by pierre01** [comment]
I have the same issue, can you explain what you did? in detail?

**Reply by pierre01** [reply]
@Cywizz  @davidortinau This is my project file what should I remove to avoid the error?:

<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<TargetFrameworks>net6.0-android;net6.0-ios;net6.0-maccatalyst</TargetFrameworks>
		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows')) and '$(MSBuildRuntimeType)' == 'Full'">$(TargetFrameworks);net6.0-windows10.0.19041</TargetFrameworks>
		<OutputType>Exe</OutputType>
		<RootNamespace>VinoMatch</RootNamespace>
		<UseMaui>true</UseMaui>
		<SingleProject>true</SingleProject>
		<ImplicitUsings>enable</ImplicitUsings>
		<EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
		
		
		<ApplicationTitle>VinoMatch</ApplicationTitle>

		
		<ApplicationId>com.companyname.VinoMatch</ApplicationId>
		<ApplicationId Condition="$(TargetFramework.Contains('-windows'))">2D1E33AC-33B1-4805-9053-559F0FB92252</ApplicationId>

		
		<ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
		<ApplicationVersion>1</ApplicationVersion>

		
		<UseInterpreter Condition="'$(Configuration)' == 'Debug'">True</UseInterpreter>

		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-ios'">14.2</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-maccatalyst'">14.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="'$(TargetFramework)' == 'net6.0-android'">21.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$(TargetFramework.Contains('-windows'))">10.0.17763.0</SupportedOSPlatformVersion>
		<TargetPlatformMinVersion Condition="$(TargetFramework.Contains('-windows'))">10.0.17763.0</TargetPlatformMinVersion>
	</PropertyGroup>

	<ItemGroup>
		
		<MauiIcon Include="Resources\appicon.svg" ForegroundFile="Resources\appiconfg.svg" Color="#512BD4" />

		
		<MauiSplashScreen Include="Resources\appiconfg.svg" Color="#512BD4" />

		
		<MauiImage Include="Resources\Images\**\*" />

		
		<MauiFont Include="Resources\Fonts\*" />
		
		
		<MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
		
	</ItemGroup>

	<ItemGroup>
	  <PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.1" />
	  <PackageReference Include="Microsoft.Toolkit.Mvvm" Version="7.1.2" />
	</ItemGroup>


	<PropertyGroup Condition="$(TargetFramework.Contains('-windows'))">
		<OutputType>WinExe</OutputType>
		<RuntimeIdentifier>win10-x64</RuntimeIdentifier>
	</PropertyGroup>

</Project>

**Comment by Cywizz** [comment]
@pierre01 , I removed the package reference to both Win2D and WindowsAppSDK...so remove this line and try again:

`<PackageReference Include="Microsoft.Graphics.Win2D" Version="1.0.1" />`
